library/Icingadb/Util/PerfDataFormat.php

```diff
@@ -0,0 +1,138 @@
+
+    protected static $bytePrefix = ['B', 'kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+    protected static $wattHourPrefix = ['Wh', 'kWh', 'MWh', 'GWh', 'TWh', 'PWh', 'EWh', 'ZWh', 'YWh'];
+
+    protected static $wattPrefix = ['W', 'kW', 'MW', 'GW', 'TW', 'PW', 'EW', 'ZW', 'YW'];
+
+    protected static $amperePrefix = ['A', 'kA', 'MA', 'GA', 'TA', 'PA', 'EA', 'ZA', 'YA'];
+
+    protected static $ampSecondPrefix = ['As', 'kAs', 'MAs', 'GAs', 'TAs', 'PAs', 'EAs', 'ZAs', 'YAs'];
+
+    protected static $voltPrefix = ['V', 'kV', 'MV', 'GV', 'TV', 'PV', 'EV', 'ZV', 'YV'];
+
+    protected static $ohmPrefix = ['O', 'kO', 'MO', 'GO', 'TO', 'PO', 'EO', 'ZO', 'YO'];
+
+    protected static $gramPrefix = ['g', 'kg', 't'];
+
+    protected static $literPrefix = ['l', 'hl'];
+    protected static $literBase = 100;
+
+    protected static $secondPrefix = ['s'];
+
+    public static function getInstance()
+    {
+        if (self::$instance === null) {
+            self::$instance = new PerfDataFormat();
+        }
+
+        return self::$instance;
+    }
+
+    public static function bits($value)
+    {
+        return self::formatForUnits($value, self::$bitPrefix, self::$generalBase);
+    }
+
+    public static function bytes($value)
+    {
+        return self::formatForUnits($value, self::$bytePrefix, self::$generalBase);
+    }
+
+    public static function wattHours($value)
+    {
+        return self::formatForUnits($value, self::$wattHourPrefix, self::$generalBase);
+    }
+
+    public static function watts($value)
+    {
+        return self::formatForUnits($value, self::$wattPrefix, self::$generalBase);
+    }
+
+    public static function amperes($value)
+    {
+        return self::formatForUnits($value, self::$amperePrefix, self::$generalBase);
+    }
+
+    public static function ampereSeconds($value)
+    {
+        return self::formatForUnits($value, self::$ampSecondPrefix, self::$generalBase);
+    }
+
+    public static function volts($value)
+    {
+        return self::formatForUnits($value, self::$voltPrefix, self::$generalBase);
+    }
+
+    public static function ohms($value)
+    {
+        return self::formatForUnits($value, self::$ohmPrefix, self::$generalBase);
+    }
+
+    public static function grams($value)
+    {
+        return self::formatForUnits($value, self::$gramPrefix, self::$generalBase);
+    }
+
+    public static function liters($value)
+    {
+        return self::formatForUnits($value, self::$literPrefix, self::$literBase);
+    }
+
+    public static function seconds($value)
+    {
+        $absValue = abs($value);
+
+        if ($absValue < 60) {
+            return self::formatForUnits($value, self::$secondPrefix, self::$generalBase);
+        } elseif ($absValue < 3600) {
+            return sprintf('%0.2f m', $value / 60);
+        } elseif ($absValue < 86400) {
+            return sprintf('%0.2f h', $value / 3600);
+        }
+
+        // TODO: Do we need weeks, months and years?
```

Comment:
I'd say, after many weeks and months and years: No.

Don't keep the comment, remove it :wink: